backends/dpdk/dpdkArch.cpp

```diff
@@ -342,6 +343,15 @@ const IR::Node *InjectJumboStruct::preorder(IR::Type_Struct *s) {
     return s;
 }
 
+const IR::Node *InjectOutputPortMetadataField::preorder(IR::Type_Struct *s) {
```

Comment:
why isn't this field part of pna.p4?

Reply:
There is no metadata field for output port in PNA spec. I don't know precisely why, the question should be rather directed to the authors of the spec. But my guess is that instead of metadata field which is in other architectures, here we have just send_to_port() extern function instead. DPDK backend needs the metadata field to store the port number internally so that it could be used for tx instruction. That's why there has to be this internally created metadata field.

We could consider different name for this metadata field though, so that it was more clear that it is not standard field defined by architecture, but field needed internally by target backend. I just kept the same name which was already used there before.